server/src/main/java/org/elasticsearch/search/diversification/ResultDiversification.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.search.diversification;
+
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.search.vectors.VectorData;
+
+import java.io.IOException;
+
+/**
+ * Base interface for result diversification.
+ */
+public abstract class ResultDiversification {
+
+    public abstract RankDoc[] diversify(RankDoc[] docs, ResultDiversificationContext diversificationContext) throws IOException;
+
+    protected float getVectorComparisonScore(
+        VectorSimilarityFunction similarityFunction,
+        boolean useFloat,
```

Comment:
Nitpick: `useFloat` as an arg feels a little weird, should we break these up e.g. something like `getFloatVectorComparisonScore` and `getByteVectorComparisonScore`